dataset.py

```diff
@@ -69,17 +74,36 @@ def _partition_data_fixed(sample_data: pd.DataFrame,
         (sample_data['lat'] >= test_bounds.min_lattitude) & (sample_data['long'] >= test_bounds.min_longitude) &
         (sample_data['lat'] < test_bounds.max_lattitude) & (sample_data['long'] < test_bounds.max_longitude)]
 
-    return PartitionedDataset(train_data, test_data, validation_data)
+    return PartitionedDataset(train=train_data, test=test_data, validation=validation_data)
+
+
+def _partition_data_random(sample_data, train_validation_test_ratios):
+    '''
+    Return sample_data split randomly into train/validation/test buckets based on
+    train_validation_test_ratios.
+    '''
+    sample_data.sample(frac=1)
+    n_train = int(sample_data.shape[0] * train_validation_test_ratios[0])
+    n_validation = int(sample_data.shape[0] * train_validation_test_ratios[1])
+
+    # Note that contrary to usual python slices, both the start and the stop are included
+    train_data = sample_data.loc[:n_train-1]
```

Comment:
@erickzul I'm pretty sure slice indexing works on a right-open interval, i.e. X[0:10] == $`X_{[0, 10)}`$. Are you sure?

Reply:
The problem was I was using loc and not iloc (the one that actually uses indexes). Updated.